src/realm/object-store/c_api/query.cpp

```diff
@@ -405,6 +419,29 @@ RLM_API realm_object_t* realm_results_get_object(realm_results_t* results, size_
     });
 }
 
+RLM_API bool realm_results_find_object(realm_results_t* results, realm_object_t* value, size_t* out_index,
+                                       bool* out_found)
+{
+    if (out_index)
+        *out_index = realm::not_found;
+    if (out_found)
+        *out_found = false;
+
+    return wrap_err([&]() {
+        if (out_index) {
+            for (size_t i = 0; i < results->size(); ++i) {
```

Comment:
Yes it is better, I did not think of doing that, there is just a small caveat though, in case an obj that does not belong to the result is searched, `index_of(obj)` will throw an exception. Which is not bad, but it will be an extra call for the SDKs to retrieve the error. 
Anyway, I changed the code.

Reply:
`index_of(obj)` throws an exception if the object you are searching for does not belong to the table, the result is based on. This is as it should be.